app/src/main/java/org/fossasia/susi/ai/data/SkillDetailsModel.kt

```diff
@@ -66,6 +68,22 @@ class SkillDetailsModel : ISkillDetailsModel {
         })
     }
 
+    override fun postFeedback(model: String, group: String, language: String, skill: String, feedback: String, listener: ISkillDetailsModel.OnUpdateFeedbackFinishedListener) {
```

Comment:
For first four parameters, use an Object. General thumb of rule, have more than three correlated paramters? Use object

Reply:
@iamareebjamal Would it be fine if I fix it along with issue #1431 ? I need to update another Pr to display feedback. I will replace all query map and queries with >3 query params with objects in the issue #1431 . Would it be fine?

Reply:
Yes, about that. Another PR got merged 5 days ago and still hasn't been resolved. This issue is related to this PR, so should be solved with it. However, if you insist, follow up PRs should be done within the next day

Reply:
@iamareebjamal I replaced it with objects. But, I am not sure how to parse this and append them as params to the URL String. I also referred to various answers on StackOverflow and found that generally the query params are sent with '@Query' or with the help of '@QueryMap'. Also, I didn't find a reason as to why is it an issue sending query params with the help of map when we send every param as String to the server. It would be helpful as if you could please help me a bit on that.

Reply:
You can send it through map in the **service** but not throughout the app. It is about limiting bad practices to the least and improving API

Reply:
Okay. Thank you @iamareebjamal , I will update the PR asap.